tests/forms_tests/tests/test_formsets.py

```diff
@@ -596,9 +596,29 @@ class Person(Form):
              'form-TOTAL_FORMS': 1, 'form-INITIAL_FORMS': 1,
              'form-MIN_NUM_FORMS': 0, 'form-MAX_NUM_FORMS': 1})
 
+        # The first form is not valid, but it can be deleted,
+        # so the formset have no errors and is valid.
         self.assertTrue(p.is_valid())
         self.assertEqual(len(p.deleted_forms), 1)
 
+    def test_formset_with_deletion_form_errors(self):
+        class Person(Form):
+            name = CharField()
+
+        PeopleForm = formset_factory(
+            form=Person,
+            can_delete=True)
+
+        p = PeopleForm(
+            {'form-0-name': '', 'form-0-DELETE': 'on',  # no name!
+             'form-TOTAL_FORMS': 1, 'form-INITIAL_FORMS': 1,
+             'form-MIN_NUM_FORMS': 0, 'form-MAX_NUM_FORMS': 1})
+
+        # The first form is not valid, but it can be deleted,
+        # so the formset have no errors and is valid.
+        self.assertTrue(p.is_valid())
+        self.assertFalse(p._errors)
```

Comment:
Adding this assertion to the previous test looks fine. It looks like the mistake before was that it was before `is_valid()` rather than after. I'd use `self.assertEqual(p._errors, {})` rather than `assertFalse`. Please rebase your branch and squash the commits.